pkg/ipam/types/types.go

```diff
@@ -333,6 +334,88 @@ type Subnet struct {
 	Tags Tags
 }
 
+// DeepEqual is a deepequal function, deeply comparing the
+// receiver with other. in must be non-nil.
+func (in *Subnet) DeepEqual(other *Subnet) bool {
+	if other == nil {
+		return false
+	}
+
+	if in.ID != other.ID {
+		return false
+	}
+	if in.Name != other.Name {
+		return false
+	}
+	if (in.CIDR == nil) != (other.CIDR == nil) {
+		return false
+	} else if in.CIDR != nil {
+		if !(&in.CIDR == &other.CIDR) {
```

Comment:
Isn't this doing a pointer comparison, even though we want a value comparison? We should use `*in.CIDR == *other.CIDR` here, no? (similar for all other fields which changed from CIDR to netip.